ArcGISRuntimeSDKQt_QMLSamples/Analysis/ConfigureSubnetworkTrace/ConfigureSubnetworkTrace.qml

```diff
@@ -0,0 +1,423 @@
+        }
+        if (traceConfiguration.traversability === null) {
+            traceConfiguration.traversability = ArcGISRuntimeEnvironment.createObject("UtilityTraversability");
+        }
+
+        const selectedNetworkAttribute = networkDefinition.networkAttribute(networkAttributeComboBox.currentText);
+        const data = convertToDataType(valueSelectionComboBox.visible ? valueSelectionComboBox.currentIndex : inputTextField.text, selectedNetworkAttribute.dataType)
+
+        if (isNaN(data)) {
+            dialogText.text = qsTr("Invalid input. Please try again.")
+            dialog.visible = true;
+            return;
+        }
+
+        // NOTE: You may also create a UtilityNetworkAttributeComparison with another NetworkAttribute.
+        const expression = ArcGISRuntimeEnvironment.createObject("UtilityNetworkAttributeComparison", {
+                                                                     networkAttribute: selectedNetworkAttribute,
+                                                                     comparisonOperator: comparisonOperatorComboBox.currentIndex,
+                                                                     value: data});
+
+        const otherExpression = traceConfiguration.traversability.barriers;
+
+        if (otherExpression) {
+            // NOTE: You may also combine expressions with UtilityTraceAndCondition
+            const combineExpressions = ArcGISRuntimeEnvironment.createObject("UtilityTraceOrCondition", {
+                                                                                 leftExpression: otherExpression,
+                                                                                 rightExpression: expression
+                                                                             });
+
+            traceConfiguration.traversability.barriers = combineExpressions;
+            expressionBuilder.text = expressionToString(combineExpressions);
+        } else {
+            traceConfiguration.traversability.barriers = expression;
+            expressionBuilder.text = expressionToString(expression);
+        }
+    }
+
+    function convertToDataType(data, dataType) {
+        switch(dataType) {
+        case Enums.UtilityNetworkAttributeDataTypeInteger:
+            return parseInt(data, 10);
+        case Enums.UtilityNetworkAttributeDataTypeFloat:
+        case Enums.UtilityNetworkAttributeDataTypeDouble:
+            return data;
+        case Enums.UtilityNetworkAttributeDataTypeBoolean:
+            return data == 1 ? true : false;
+        }
+    }
+
+    function reset() {
+        traceConfiguration.traversability.barriers = initialExpression;
+        expressionBuilder.text = expressionToString(initialExpression);
+    }
+
+    function trace() {
+        busyIndicator.visible = true;
+        if (!utilityNetwork || !utilityElementStartingLocation) {
+            return;
```

Comment:
Same comment as C++. If you return early here, `busyIndicator.visible = true;` is probably wrong. You may want to put that in the else block.